src/network_models/network_constructor.jl

```diff
@@ -6,13 +6,28 @@ function construct_network!(
     buses = PSY.get_components(PSY.Bus, sys)
     bus_count = length(buses)
 
-    get_slack_variables(psi_container.settings) &&
+    get_balance_slack_variables(psi_container.settings) &&
         add_slacks!(psi_container, CopperPlatePowerModel)
     copper_plate(psi_container, :nodal_balance_active, bus_count)
 
     return
 end
 
+function construct_network!(
+    psi_container::PSIContainer,
+    sys::PSY.System,
+    ::Type{AreaBalancePowerModel},
+)
+    area_mapping = PSY.get_aggregation_topology_mapping(PSY.Area, sys)
+    branches = get_available_components(PSY.Branch, sys)
+    if get_balance_slack_variables(psi_container.settings)
+        @warn("Slack Variables are not compatible with AreaBalancePowerModel")
```

Comment:
Maybe this has already been covered; please delete if that is the case.  Should this call `error()` instead?